87.md

```diff
@@ -0,0 +1,129 @@
+    * Queries for `kind:18173` and, based on results, queries for `kind:38173`/`kind:38172`
+
+## Events
+
+### Recommendation event
+```json
+{
+  "kind": 18173,
+  "pubkey": <recommender-user-pubkey>,
+  "tags": [
+    ["u", <recommended-cashu-mint-url>, "cashu"],
+    ["u", <recommended-fedimint-invite-code>, "fedimint"],
+    ["u", <recommended-cashu-mint-url>, "cashu"],
+    ["a", "38173:fedimint-pubkey:<d-identifier>", "wss://relay1", "fedimint"],
+    ["a", "38172:cashu-mint-pubkey:<d-identifier>", "wss://relay2", "cashu"]
+  ],
+  "content": ""
+}
+```
+
+`u` tags are used to point directly to the ecash mint.
+The first value of the tag is the URL or invite code of the ecash mint, and the second value is the type of ecash mint (currently `cashu` or `fedimint`).
+Multiple `u` tags can appear on the same `kind:18173`.
+
+`a` tags are used to point to the `kind:38173`/`kind:38172` event of the ecash mint.
+The first value of the tag is the `kind:38173`/`kind:38172` event identifier, the second value of the tag is a relay hint, and the third value of the tag is the type of ecash mint (currently `cashu` or `fedimint`).
+
+## Ecash Mint Information
+
+Cashu mints SHOULD publish `kind:38172` events to announce their capabilities and how to connect to them.
+
+For cashu mints, the `u` tag SHOULD be the URL to the cashu mint and it should list the nuts that the cashu mint supports.
+
+The `d` tag SHOULD be the mint's pubkey (found when querying `/v1/info`), this way users can query by pubkey and find the mint announcement.
+
+```json
+{
+  "kind": 38172,
+  "pubkey": "<application-pubkey>",
+  "content": "<optional-kind:0-style-metadata>",
+  "tags": [
+    ["d", <cashu mint pubkey>],
+    ["u", "https://cashu.example.com"],
+    ["nuts", "1,2,3,4,5,6,7"]
+  ]
+}
+```
+
+Fedimints SHOULD publish `kind:38173` events to announce their capabilities and how to connect to them.
+
+For fedimints, it should list all known fedimint invite codes in `u` tags and it should list the modules it supports.
+
+The `d` tag SHOULD be the federation id, this way users can query by federation id and find the fedimint announcement.
+
+```json
+{
+  "kind": 38173,
+  "pubkey": "<application-pubkey>",
+  "content": "<optional-kind:0-style-metadata>",
+  "tags": [
+    ["d", <federation-id>],
+    ["u", "fed11abc.."],
+    ["u", "fed11xyz.."],
+    ["modules", "lightning,wallet,mint"]
+  ]
+}
+```
+
+* `content` is an optional `metadata`-like stringified JSON object, as described in NIP-01. This content is useful when the pubkey creating the `kind:38173` is not a normal user. If `content` is empty, the `kind:0` of the pubkey should be used to display mint information (e.g. name, picture, web, LUD16, etc.)
+
+## Example
+
+### User A recommends some mints
+User A might be a user of a cashu mint. Using a client, user A publishes an event recommending the cashu mint they use.
+
+```json
+{
+  "kind": 18173,
+  "tags": [
+    ["u", "fed11abc..", "fedimint"],
```

Comment:
Why would the recommendation contain the full invite code? Feels like the federation ID would be sufficient since in the second step the user client would query the 38172/38173 event anyway?

Reply:
> Why would the recommendation contain the full invite code? Feels like the federation ID would be sufficient since in the second step the user client would query the 38172/38173 event anyway

my main reasoning was so that you can recommend a federation even if they never announced themselves. Ideally every federation does but at least for now this is a manual process and to bootstrap this it'd be easier to support both options.

Reply:
Ah, the federation is supposed to announce itself? I thought the recommending user would do that if not the case already.